### ISSUE TITLE
Linter messages not always visible

### ISSUE THREAD

**Issue by LarsMichelsen** [neoclide/coc-python#29]
After starting neovim with current version of coc+coc-python I always get the correct information from the linter, can cycle through the messages with coc-diagnostic-next and get the detail message in the popup next to the cursor.

From time to time the popup does not work anymore and I have no chance to see the detail message of the linter. The icons on the side of the lines are correctly created. And the messages are visible with ":CocCommand python.viewOutput".

a) Would love to have this working without issues.
b) In case it is not easy fixable, would it be possible to show the message in the command-line?

Is there anything needed I can send you to help?

**Comment by chemzqm** [comment]
Make sure `echo exists('*nvim_open_win')` is `1`.

**Comment by LarsMichelsen** [comment]
The call always returns 1.

May be interesting: The completion popups work all the time, only the linter messages are not shown. This does not seem to be a general popup window issue.

I also recognized that the "_" marker on the lines with the linter issues are not visible.

**Comment by chemzqm** [comment]
The diagnostics you have should comes from other plugin like ALE which enables all linters for you without any configuration.
Run `:CocList diagnostics` to get list of diagnostics from coc.

**Comment by LarsMichelsen** [comment]
`:CocList diagnostics` is showing the messages just fine. Only the popups break from time to time

This is my current coc-settings.json:

```
{
    // https://github.com/neoclide/coc.nvim/blob/master/data/schema.json<Paste>
    "suggest.enablePreview": true,
    "suggest.triggerAfterInsertEnter": true,
    //"suggest.echodocSupport": true,
    "signature.enable": true,

    "diagnostic.refreshOnInsertMode": true,
    "diagnostic.errorSign": "\uf05e",
    "diagnostic.infoSign": "\uf071",
    "coc.preferences.formatOnSaveFiletypes": ["python"],

    "python.pythonPath": "${workspaceFolder}/.venv/bin/python",
    "python.autoComplete.extraPaths": [
        "${workspaceFolder}/livestatus/api/python",
    ],
    "python.autoComplete.typeshedPaths": [],
    "python.venvFolders": [".venv"],
    "python.venvPath": "${workspaceFolder}",

    "python.analysis.disabled": ["undefined-variable"],
    "python.jediEnabled": false,

    "python.linting.pylintEnabled": true,
    "python.linting.pylintPath": "${workspaceFolder}/.venv/bin/pylint",
    "python.linting.pylintUseMinimalCheckers": false,

    "python.formatting.provider": "yapf",

    "python.linting.mypyEnabled": true,
    "python.linting.mypyPath": "${workspaceFolder}/tests/static/.venv/bin/mypy",
    "python.linting.mypyArgs": [
        "--ignore-missing-imports",
        "--follow-imports=silent",
        "--show-column-numbers",
        "--python-version 2.7",
    ],

    "python.linting.banditEnabled": true,
    "python.linting.banditPath": "${workspaceFolder}/.venv/bin/bandit",

    // Some optional debugging to get more information via ":CocCommand workspace.showOutput"
    "python.trace.server": "verbose",
}

```

**Comment by chemzqm** [comment]
>  Only the popups break from time to time.

Could be caused by your other vim plugin, it should be shown after 500ms cursor moved to it.

**Comment by LarsMichelsen** [comment]
I've removed all vim plugins except coc but that did not change anything.

I noticed another thing: I can save a file without changes as often as I want and the popups still work.
But once I add a new line to the file the popups break. Removing a line from the file does not break it.

In the meantime I tried to solve this by using `"diagnostic.messageTarget": "echo"`, but the messages also break in this mode.

Looks like the line matching breaks somehow.

...

Ok, one step closer: I had `"diagnostic.refreshOnInsertMode": true`. Once I disable it, the popups keep working.
